### PR TITLE
close connection in case of DatabaseError

### LINKED ISSUE #130: easyaccess unresponsive after Oracle error
I'm running easyaccess using the docker container on the docker hub.

If I make a syntax error in a query, and get an error message from Oracle (printed in red by easyaccess), it returns me to the prompt but the subsequent command will not run, and the tool becomes unresponsive. I have to kill the docker container and start again.

To reproduce:

```
DESDB ~> select count(blah from spod;

    |   ☆          |  Ctrl-C to abort; 
<class 'cx_Oracle.DatabaseError'>
ORA-00907: missing right parenthesis

DESDB ~> [Anything such as a query or 'exit' command - hangs]
```

### PR DESCRIPTION
**Pull request by madamow**
Fixes #130 
@mgckind PTAL
Code catches the cx_Oracle.Database error due to incorrect syntax in query and closes the connection instead of trying to cancel it.
Cancelling connection causes it to hang, closing it allows to go around it.

### CODE REVIEW

**Comment by mgckind** [comment]
😱, that was quick. Thanks, I'll take a look

File: easyaccess/easyaccess.py [hunk 1]
```diff
@@ -801,7 +801,25 @@ def query_and_print(self, query, print_time=True,
...
+            # close the connection to the database instead of trying to cancel a transaction.
+            # Otherwise, the connection will hang indefinitely during the next attempt to utilize it.
+            # See also:
+            # * https://community.oracle.com/thread/3612149
+            # * https://community.oracle.com/thread/717738
+            # * https://github.com/mgckind/easyaccess/issues/130
+            print(colored('Your query raised DatabaseError:', "red", self.ct))
+            print(colored(str(exc), "red", self.ct))
+            self.con.close()
+            if self.loading_bar:
```
Comment (mgckind) [inline easyaccess/easyaccess.py:815]: I think I have it somewhere, but is the connection reopen after closing it?

File: easyaccess/easyaccess.py [hunk 2]
```diff
@@ -801,7 +801,25 @@ def query_and_print(self, query, print_time=True,
...
+                if self.pload.pid is not None:
+                    os.kill(self.pload.pid, signal.SIGKILL)
+        except Exception as exc:
+            (type, value, traceback) = sys.exc_info()
+            self.con.cancel()
+            t2 = time.time()
+
             (type, value, traceback) = sys.exc_info()
             self.con.cancel()
```
Comment (mgckind) [inline easyaccess/easyaccess.py:824]: aren't these lines duplicated? they don't show are being removed

File: easyaccess/easyaccess.py (hunk 1 above)
Comment (madamow) [reply easyaccess/easyaccess.py:815]: As far as I checked the next step after closing/canceling connection is pinging the database to check the connection. If the ping comes back as False, the code will connect to database again. Try to run invalid query and then a valid one. 

https://github.com/madamow/easyaccess/blob/318357fbb2f1a56819d16bdc59d1a6032fb30e0f/easyaccess/easyaccess.py#L438-L451

Comment (mgckind) [reply easyaccess/easyaccess.py:815]: oh i see, yes, that's what i meant. great!

**Review by mgckind — APPROVED** [review]

**Pull request merged**